sys/src/9/amd64/devvq.c

```diff
@@ -0,0 +1,638 @@
+/*
```

Comment:
I don't see anything about virtio that is architecture-specific. Note that rusty russell had a lot to do with virtio and at that time he was an IBM Power guy.